### DISCUSSION TITLE
state estimates ramping and resetting

### DISCUSSION THREAD

**Discussion by jjdike** [bitcraze/discussions#1160] [Q&A] [answered]
Hi - I am just getting going with crazyflie and have been able to connect to the copter with cfclient.  I am seeing the x and y state estimates continually ramp up (x ramps down, y ramps up) till one of them hits 100 then the states reset - while the crazyflie is just sitting on the table.  Z stays at about 0, even if I move the copter around z doesn't change.  The roll, pitch, yaw numbers seem reasonable.  This is with the  flow deck installed.  In case it is useful, here's the console trace from connecting:

SYS: Crazyflie 2.1 is up and running!
SYS: Build 0:e527895e8c6e (2023.11) CLEAN
SYS: I am 0x33373336343247060018002E and I have 1024KB of flash!
CFGBLK: v1, verification [OK]
DECK_CORE: 1 deck(s) found
DECK_CORE: Calling INIT on driver bcFlow2 for deck 0
ZR2: Z-down sensor [OK]
PMW: Motion chip id: 0x49:0xB6
IMU: BMI088: Using I2C interface.
IMU: BMI088 Gyro connection [OK].
IMU: BMI088 Accel connection [OK]
IMU: BMP388 I2C connection [OK]
ESTIMATOR: Using Kalman (2) estimator
CONTROLLER: Using PID (1) controller
MTR-DRV: Using brushed motor driver
SYS: About to run tests in system.c.
SYS: NRF51 version: 2023.11
EEPROM: I2C connection [OK].
STORAGE: Storage check [OK].
IMU: BMI088 gyro self-test [OK]
DECK_CORE: Deck 0 test [OK].
SYS: Self test passed!
STAB: Wait for sensor calibration...
SYS: Free heap: 17608 bytes
STAB: Starting stabilizer loop
ESTKALMAN: State out of bounds, resetting
SUP: Can not fly
SUP: Ready to fly

A couple of other questions that I can split off from this are that the connection does not last long - maybe a minute is my longest connection before it disconnects with a 'can't send packets to copter' message.  I've found the most reliable way to reconnect is to pull the dongle out and back in.  

Thank you.

**Comment by knmcguire** [comment]
Hi! this diverging can be due to 2 things I can think off:
- The flowdeck is looking at a very featureless or black table
- The flowdeck is mounted too close to the table

We actually had a blogpost and a devmeeting about these issues occurring with the flowdeck so perhaps that helps:
- blogpost: https://www.bitcraze.io/2023/11/go-with-the-flow-relative-positioning-with-the-flow-deck/
- devmeeting thread with recording: https://github.com/orgs/bitcraze/discussions/1056

For your crazyradio (version 2?) question, that seems to be unrelated to the flow so please make a new thread for that.

**Reply by jjdike** [reply, marked as answer]
Ok so after looking at your blog post for a second and getting the sense of what the flow deck is doing I realized the problem.  I had put the flowdeck on top fo the battery, with the sensor facing the battery!  I put the flow deck below the crazyflie with the sensor facing the floor and it hovers in assist mode!

What threw me off, besides not really thinking about what the flow card was supposed to be doing, was starting from the flow deck v2 purchase page, going to the 'get started' link, then the 'getting started with expansion boards' page, was just looking at the board mounting pictures there and initially thinking it was showing various optional ways to mount the decks.  I'd suggest possibly putting a note on that page that certain boards need to be located in particular mounting locations to work properly, even if it should be obvious that a board that is sensing height should have the height sensor facing down.  In the 'getting started with the flow deck' help page, https://www.bitcraze.io/documentation/tutorials/getting-started-with-flow-deck/, there is a picture near the bottom of the page that shows the orientation of the flow board.  I remember looking at that picture but it didn't register that the board location mattered, just thinking of the images on the 'getting started with expansion boards' page.  I think because there are a lot of different and somewhat related help pages it may be easy to miss some relevant information for those going through the setup for the first time.  Anyway this is user error but just some info on where my initial confusion came from.

Thanks again for the daily help with this.

**Reply by knmcguire** [reply]
Ahhh yes, that won't work indeed. We will keep it in mind to clarify the documentation in case more people experience this issue, so thanks for the feedback.
